core/tauri-runtime-wry/src/lib.rs

```diff
@@ -3246,13 +3246,35 @@ fn create_webview<T: UserEvent>(
   if window_builder.center {
     let _ = center_window(&window, window.inner_size());
   }
-  let mut webview_builder = WebViewBuilder::new(window)
-    .map_err(|e| Error::CreateWebview(Box::new(e)))?
-    .with_focused(focused)
-    .with_url(&url)
-    .unwrap() // safe to unwrap because we validate the URL beforehand
-    .with_transparent(is_window_transparent)
-    .with_accept_first_mouse(webview_attributes.accept_first_mouse);
+
+  let url_str = url.to_string();
+
+  let mut webview_builder =
+      WebViewBuilder::new(window).map_err(|e| Error::CreateWebview(Box::new(e)))?;
+
+  // use with_html method if html content can be extracted from url.
+  // else defaults to with_url method
+  webview_builder = if let Some(html_string) = tauri_utils::html::extract_html_content(&url_str) {
+      // we run a decode here on the html_string for sanity check
+      // the content should be fully decoded when passing to WRY, which
+      // will be responsible for handling the encoding based on the OS.
+      let html =
```

Comment:
I don't think `tauri-runtime-wry` should handle this case, it should be handled by `tauri` crate and passed as a valid url.

Reply:
I feel like this should be the responsibility of `wry` instead of `tauri `or even `tauri-runtime-wry`, as it is dependent on the underlying web engine on each platform on how the html string is parsed. For windows, `wry `does the following to render html content on webview2:

```
      unsafe {
        webview
          .NavigateToString(PCWSTR::from_raw(encode_wide(html).as_ptr()))
          .map_err(webview2_com::Error::WindowsError)?;
      }
```

which encodes the HTML content. So, by passing in an already encoded uri, it will result in a double encoding. This is not an issue on linux where it seems to handle this issue by default.

I am rather unfamiliar with the underlying technologies, and this is based on my rudimentary understanding from looking into this issue thus far. Would greatly appreciate any corrections or insights on this.